lib/internal/url.js

```diff
@@ -839,20 +883,26 @@ class URL {
     this.#context.hash_start = hash_start;
     this.#context.scheme_type = scheme_type;
 
-    if (this.#searchParams) {
-      if (this.#context.hasSearch) {
-        setURLSearchParams(this.#searchParams, this.search);
+    // If URLSearchParams exists, and the search has changed, update it.
+    // Otherwise, check if URLSearchParams has been modified.
+    const currentSearch = updateSearchParams && this.#getSearchFromContext();
+    if (updateSearchParams) {
```

Comment:
Would it make sense to skip the `currentSearch` initialization if `updateSearchParams` is falsy?
```suggestion
    if (updateSearchParams) {
      const currentSearch = updateSearchParams && this.#getSearchFromContext();
```

Reply:
Yah, and the `updateSearchParams` in the constant would no longer be needed either:

```suggestion
    if (updateSearchParams) {
      const currentSearch = this.#getSearchFromContext();
```